From 2560e71132b1ad0835655c55821ca70c1a9968c2 Mon Sep 17 00:00:00 2001
From: Liam Girdwood <liam.r.girdwood@linux.intel.com>
Date: Fri, 3 Nov 2017 20:56:42 +0000
Subject: [PATCH 1211/1240] ALSA: HACK: Fix rmmod crash

Something appears to corrupt the runtime->status on second playback...

Signed-off-by: Liam Girdwood <liam.r.girdwood@linux.intel.com>
---
 sound/core/pcm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/core/pcm.c b/sound/core/pcm.c
index e8dc1a5..10cdc0d 100644
--- a/sound/core/pcm.c
+++ b/sound/core/pcm.c
@@ -1158,6 +1158,7 @@ static int snd_pcm_dev_disconnect(struct snd_device *device)
 	list_del_init(&pcm->list);
 	for (cidx = 0; cidx < 2; cidx++) {
 		for (substream = pcm->streams[cidx].substream; substream; substream = substream->next) {
+#if 0
 			snd_pcm_stream_lock_irq(substream);
 			if (substream->runtime) {
 				substream->runtime->status->state = SNDRV_PCM_STATE_DISCONNECTED;
@@ -1165,6 +1166,7 @@ static int snd_pcm_dev_disconnect(struct snd_device *device)
 				wake_up(&substream->runtime->tsleep);
 			}
 			snd_pcm_stream_unlock_irq(substream);
+#endif
 		}
 	}
 	if (!pcm->internal) {
-- 
2.7.4

